src/coreclr/src/jit/optcse.cpp

```diff
@@ -3280,10 +3280,25 @@ bool Compiler::optIsCSEcandidate(GenTree* tree)
         case GT_LE:
         case GT_GE:
         case GT_GT:
-            return true; // Also CSE these Comparison Operators
+            return true; // Allow the CSE of Comparison operators
+
+#ifdef FEATURE_SIMD
+        case GT_SIMD:
+            return true; // allow SIMD intrinsics to be CSE-ed
+
+#endif // FEATURE_SIMD
+
+#ifdef FEATURE_HW_INTRINSICS
+        case GT_HWINTRINSIC:
+            return true; // allow Hardware Intrinsics to be CSE-ed
```

Comment:
Are there any HWIntrinsics we don't want CSE'd or any that need special handling (such as the non-temporal load/stores)?

Reply:
Excellent thought - I would say this should include:
- any of the "special" loads (`LoadFence`, any `NonTemporal`
- any store (i.e. `HW_Category_MemoryStore` and `StoreFence`)
- any prefetch
- `MemoryFench`

Perhaps we should consider adding a flag for this, but a reasonable proxy in the meantime would be to exclude any `HW_Category_MemoryStore`, `HW_Category_MemoryLoad` and `HW_Category_Special`:
- There would be few, if any, loads that the JIT could identify as CSE's (local stack loads would generally not show up as actual load intrinsics).
- The only `HW_Category_Special` that I found that this would unnecessarily exclude is `CompareLessThan`